pyccel/ast/variable.py

```diff
@@ -738,18 +699,45 @@ def is_target(self, is_target):
             if var.rank > 0:
                 var.is_target = is_target
 
-class Constant(ValuedVariable):
+class Constant(Variable):
 
     """
+    Class for expressing constant values (e.g. pi)
+
+    Parameters
+    ----------
+    *args, **kwargs : See pyccel.ast.variable.Variable
+
+    value : Type matching dtype
+            The value that the constant represents
 
     Examples
     --------
+    >>> from pyccel.ast.variable import Constant
+    >>> import math
+    >>> Constant('real', 'pi' , value=math.pi )
+    Constant('pi', dtype=NativeReal())
 
     """
-    __slots__ = ()
+    __slots__ = ('_value',)
     # The value of a constant is not a translated object
     _attribute_nodes = ()
 
+    def __init__(self, *args, value = Nil(), **kwargs):
+        self._value = value
+        super().__init__(*args, **kwargs)
+
+    @property
+    def value(self):
+        """ Default value of the variable
+        """
+        return self._value
```

Comment:
I don't think this docstring properly describes the meaning of the `value` property. I'd rather write `Immutable value of the constant`

Reply:
Fixed 191b08e04c129b132df197a830b071860d6233d3